src/main/java/frc/robot/subsystems/TrapElvSubsystem.java

```diff
@@ -0,0 +1,374 @@
+
+    baseMotor2 = new CANSparkMax(TrapElvConstants.baseMotor2_ID, MotorType.kBrushless);
+    baseMotor2.restoreFactoryDefaults();
+    baseMotor2.getPIDController().setP(basePID[0]);
+    baseMotor2.getPIDController().setI(basePID[1]);
+    baseMotor2.getPIDController().setD(basePID[2]);
+    baseMotor2.getPIDController().setIZone(basePID[3]);
+    baseMotor2.getPIDController().setFF(basePID[4]);
+
+    baseLimit = new DigitalInput(TrapElvConstants.baseBreak_ID);
+
+    scoringMotor = new CANSparkMaxSim(TrapElvConstants.scoringMotor_ID, MotorType.kBrushless);
+    scoringMotor.restoreFactoryDefaults();
+    scoringMotor.getPIDController().setP(scoringPID[0]);
+    scoringMotor.getPIDController().setI(scoringPID[1]);
+    scoringMotor.getPIDController().setD(scoringPID[2]);
+    scoringMotor.getPIDController().setIZone(scoringPID[3]);
+    scoringMotor.getPIDController().setFF(scoringPID[4]);
+    TrapElvTab.add("Scoring Elv PID", scoringMotor.getPIDController());
+
+    scoringLimit = new DigitalInput(TrapElvConstants.scoringBreak_ID);
+
+    // SmartDashboard
+    sourceLog = new DebugEntry<Boolean>(baseLimit.get(), "Source Beam Break", this);
+    groundLog = new DebugEntry<Boolean>(baseLimit.get(), "Ground Beam Break", this);
+    baseLog = new DebugEntry<Boolean>(baseLimit.get(), "Base Limit Switch", this);
+    scoringLog = new DebugEntry<Boolean>(baseLimit.get(), "Scoring Limit Switch", this);
+
+    // Simulation
+    if (Robot.isSimulation()) {
+      // 2D Mechanism
+      elvMechanism = new Mechanism2d(2, 2);
+      root = elvMechanism.getRoot("Root", 1, 0);
+      baseMech =
+          root.append(new MechanismLigament2d("Base Arm", 0, 90, 20, new Color8Bit(Color.kBlue)));
+      scoringMech =
+          baseMech.append(
+              new MechanismLigament2d("Scoring Arm", 0, 0, 10, new Color8Bit(Color.kAqua)));
+      wristMech =
+          scoringMech.append(
+              new MechanismLigament2d(
+                  "Wrist Mech", TrapElvConstants.wristLength, -170, 5, new Color8Bit(Color.kRed)));
+
+      m_baseElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.elevatorGearRatio,
+              TrapElvConstants.elevatorCariageMass,
+              Units.inchesToMeters(1),
+              TrapElvConstants.elevatorMinHight,
+              TrapElvConstants.elevatorMaxHight,
+              true,
+              0);
+
+      m_scoringElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.elevatorGearRatio,
+              TrapElvConstants.elevatorCariageMass,
+              Units.inchesToMeters(1),
+              TrapElvConstants.elevatorMinHight,
+              TrapElvConstants.elevatorMaxHight,
+              true,
+              0);
+
+      m_wristMotorSim =
+          new SingleJointedArmSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.elevatorGearRatio,
+              1,
```

Comment:
Not certain about the physics, but I think the moment of inertia is:
(1/3) * (wristLength^2) * elevatorCariageMass

Mike said most of the weight is the wrist arm, so using the same elevator mass seems reasonable. 

1 kg\*m^2 is actually much higher than this formula expects (~0.18 kg*m^2)